ssl/record/rec_layer_s3.c

```diff
@@ -306,7 +306,7 @@ int ssl3_read_n(SSL *s, int n, int max, int extend, int clearold)
             if (s->mode & SSL_MODE_RELEASE_BUFFERS && !SSL_IS_DTLS(s))
                 if (len + left == 0)
                     ssl3_release_read_buffer(s);
-            return (i);
+            return -1;
```

Comment:
It gets propagated all the way up to the return value of SSL_read() which does document what 0 means and what <0 means, and they are different.